src/main/java/frc/robot/subsystems/BallPath/Shooter/PIDShooterTrackingImpl.java

```diff
@@ -0,0 +1,363 @@
+        if(aim){
+            // only get the distance if we are aiming
+            tx = table.getEntry("tx");
+            ty = table.getEntry("ty");
+            ta = table.getEntry("ta");
+            tv = table.getEntry("tv");
+            x = tx.getDouble(0.0);
+            y = ty.getDouble(0.0);
+            a = ta.getDouble(0.0);
+            canSeeTarget = tv.getDouble(0.0);
+            totalAngle = a1+y;
+            totalAngleRadians = Math.toRadians(totalAngle);
+            rs = Math.tan(totalAngleRadians);
+            totalDistance = heightDif / rs;
+            
+            if(turretEncoderReadingPosition > leftLimit && turretEncoderReadingPosition < rightLimit && !flipLeft && !flipRight){
+                if(canSeeTarget==1.0){
+                    // if we are in the encoder limits, can see the target, and do not want to "flip"
+                    if(x < 1 && x > -1){
+                        turretMotor.set(ControlMode.PercentOutput, 0);
+                        turretReady = true;
+                    }else if(x > 1){
+                        turretMotor.set(ControlMode.PercentOutput, turretSpeed);
+                        turretReady = false;
+                    }else if(x<-1){
+                        turretMotor.set(ControlMode.PercentOutput, -turretSpeed);
+                        turretReady = false;
+                    }
+                }else{
+                    // if we cannot see the target, flip to one of the limits
+                    if(turretEncoderReadingPosition >= 0){
+                        flipLeft = true;
+                        turretReady = false;
+                    }else if(turretEncoderReadingPosition < 0){
+                        flipRight = true;
+                        turretReady = false;
+                    }
+                }
+            // if we are past the left limit, flip to the right
+            }else if(turretEncoderReadingPosition < leftLimit || flipRight == true){
+                flipRight = true;
+                turretReady = false;
+                if(turretEncoderReadingPosition < rightLimit - turretBuffer && canSeeTarget == 0.0){
+                    turretMotor.set(ControlMode.PercentOutput, turretSpeed);
+                }else{
+                    flipRight = false;
+                }
+            // if we are at the right limit, flip to the left
+            }else if(turretEncoderReadingPosition > rightLimit || flipLeft == true){
+                turretReady = false;
+                flipLeft = true;
+                if(turretEncoderReadingPosition > leftLimit + turretBuffer && canSeeTarget == 0.0){
+                    turretMotor.set(ControlMode.PercentOutput, -turretSpeed);
+                }else{
+                    flipLeft = false;
+                }
+            }    
+        // if we want to shoot the fender shot, set the turret position to be 0 (straight)
+        // this may change if we have a turret that can turn 180 degrees
```

Comment:
This comment seems misleading. It's talking about setting the turret to the forward position, but the code below is about setting the hood height?